apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -64,10 +64,11 @@
     id := emqx_ds:session_id(),
     created_at := _Millisecond :: non_neg_integer(),
     expires_at := _Millisecond :: non_neg_integer() | never,
-    iterators := map(),
     props := map()
 }.
 
+-type iterators() :: #{topic() => iterator()}.
```

Comment:
Strictly speaking, we can not rely on types defined in the `emqx` app since it's not a dependency. Refined the type in 2a1e0ad48. Was unsure about renaming, I have a feeling that on this level `topic()` reflects the nature of what it is better than `words()`, which is essentially a name for specific representation.

Reply:
Fair enough, we can't reference the `emqx` app indeed.

Though `list(binary())` might be slightly imprecise, since we end up using `emqx_topic:words/1` to create those values, which introduce the atoms cited above.  Perhaps we could reflect that in this module's type?

```erlang
-type topic() :: list(binary() | '' | '+' | '#').
```

Reply:
Just saw this change in another commit.  :smile_cat:

Reply:
Absolutely! This one is now called `topic_filter()` after 35e91ac.